File: frontend/src/api/dto/boundsDto.ts
/**
 * Geographic bounding box, in WGS84 degrees
 */
export interface BoundsDto {
  /** Western edge */
  minLon: number
  /** Southern edge */
  minLat: number
  /** Eastern edge */
  maxLon: number
  /** Northern edge */
  maxLat: number
}

File: backend/src/main/java/fr/pedalons/dto/routes/response/BoundsDto.java
package fr.pedalons.dto.routes.response;

import fr.pedalons.dto.validation.ValidateSchema;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Geographic bounding box, in WGS84 degrees")
@ValidateSchema
public record BoundsDto(
    @Schema(description = "Western edge", required = true) double minLon,
    @Schema(description = "Southern edge", required = true) double minLat,
    @Schema(description = "Eastern edge", required = true) double maxLon,
    @Schema(description = "Northern edge", required = true) double maxLat) {}
